common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleService.java

```diff
@@ -416,7 +416,7 @@ public void initializeApplication(ApplicationInitializationContext context) {
     String appId = context.getApplicationId().toString();
     try {
       ByteBuffer appServiceData = context.getApplicationDataForService();
-      String payload = JavaUtils.bytesToString(appServiceData);
+      String payload = JavaUtils.bytesToString(appServiceData.slice());
```

Comment:
This was an actual bug, that it used to work was luck - the . The buffer is consumed twice, here and later on when we serialize the password. In that case the second serialization will read an exhausted buffer.